Create rule S7165: New Relic secrets should not be disclosed (APPSEC-2253) (#4503)
SonarSource/rspec: rules/S7165/secrets/rule.adoc

include::../../../shared_content/secrets/description.adoc[]

== Why is this an issue?

include::../../../shared_content/secrets/rationale.adoc[]

=== What is the potential impact?

> If attackers gain access to a New Relic API key, they might be able to gain access to inject forged data,
> to retrieve information potentially private or to configure features in New Relic monitoring solutions.

Reviewer: It is difficult for me to read this line, I suggest

```suggestion
If attackers gain access to a New Relic API key, they could potentially inject forged data, retrieve sensitive information, or configure features within New Relic monitoring solutions.
```

> Below are some real-world scenarios that illustrate some impacts of an attacker
> exploiting the secret..

Reviewer: ```suggestion
Below are some real-world scenarios that illustrate some impacts of an attacker
exploiting the secret.
```

include::../../../shared_content/secrets/impact/data_compromise.adoc[]

include::../../../shared_content/secrets/impact/data_modification.adoc[]

== How to fix it

include::../../../shared_content/secrets/fix/revoke.adoc[]

include::../../../shared_content/secrets/fix/recent_use.adoc[]

include::../../../shared_content/secrets/fix/vault.adoc[]

=== Code examples

> :example_secret: NRAK-S6Z8F90IQDWRILXAKBBESCDIY99

Reviewer: This key is not currently present in the sonar-text detection logic, I suggest adding it there to ensure this would raise an issue
